src/grand-search/gui/searchconfig/searchenginewidget.cpp

```diff
@@ -137,3 +160,8 @@ int SearchEngineWidget::getIndex(const QString text) const
     }
     return index;
 }
+
+void SearchEngineWidget::setCustomSearchEngineAddress(QString text)
+{
+    SearchConfig::instance()->setConfig(GRANDSEARCH_WEB_GROUP, GRANDSEARCH_WEB_SEARCHENGINE_CUSTOM_ADDR, text);
```

Comment:
please encode the input url by base64.

Reply:
I can't understand why I need to use the base64 encoding

Reply:
To avoid that the string that contains some Illegal characters corrupts the configuration file.